src/components/walletconnect/SessionManager/index.tsx

```diff
@@ -61,6 +64,13 @@ const SessionManager = ({ sessions }: { sessions: SessionTypes.Struct[] }) => {
     return walletConnect.onSessionPropose(setProposal)
   }, [walletConnect])
 
+  // Connect to session present in URL
+  useEffect(() => {
+    if (!walletConnect || !wcUri) return
+
+    walletConnect.connect(wcUri).catch(setError)
```

Comment:
I would put stuff like that in the context, not in the session manager component.